src/services/rfqm_service.ts

```diff
@@ -573,6 +574,29 @@ export class RfqmService {
         }
     }
 
+    /**
+     * Runs checks to determine the health of the RFQm system. The results may be distilled to a format needed by integrators.
+     */
+    // tslint:disable-next-line prefer-function-over-method
+    public async runHealthCheckAsync(): Promise<HealthCheckResult> {
+        const transformPairs = (offerings: RfqMakerAssetOfferings): { [pair: string]: HealthCheckStatus } =>
```

Comment:
can we define this outside the function?

Reply:
Do you think that's better?

I think some engineers would see this as odd, but that's because so much of the industry is traumatized by Java (see ["Execution in the Kingdom of Nouns" rant](https://steve-yegge.blogspot.com/2006/03/execution-in-kingdom-of-nouns.html). If you're from that world you'd certainly make this a method, but that's because you have no other choice. Since `runHealthCheckAsync` is the only function concerned with the `transformPairs` logic, why move it to a broader context when no other member of that context cares about it? My inclination is actually the opposite: make it an IIFE and define and invoke it at the same time. But my sense is that this is even more odd to folks.

Reply:
I find it to be more readable personally. You can more quickly see what `runHealthCheckAsync` is doing (returning operational and applying a transformation to the pairs), and you might not care about digging into the `transformPairs` logic--but if you do then you can check out the function separately.

If we define the logic for every utility `runHealthCheckAsync` needs within the function itself, it could get crowded and hard to follow.

Reply:
I forgot to press submit, but here's my two cents:

There are other contexts where this could be helpful (to short circuit certain code paths). Additionally, since this is pulling in an environment variable that won't change until a restart, we can pay the compute cost once on startup - as opposed to once per request. My preference would be to have this be done once on startup, and stored in memory as a singleton to be reused across contexts. 

We can do this later thought

Reply:
This is one of those great coding debates that we can go back and forth on forever. It is my hope that once I am older and grey I will have my reasonings so well ironed out I can convince anybody in like five minutes.

The one note I'll add here is the env variable already has the behavior you want @phil-ociraptor. That config file grabs it once and that's that.